### ISSUE TITLE
`train.py` lacks deterministic seed handling

### ISSUE THREAD

**Issue by gelluisaac** [Traqora/astroml#159]
Description:
Training scripts don't expose seeds consistently causing non-reproducible results.

Suggested fix:
- Add `--seed` flag and set seeds for numpy, random, torch.

---

**Comment by oscarj007** [comment]
@oscarj007 has applied to work on this issue as part of the **Stellar Wave Program**'s 5th wave.



> Permit me to solve this issue please 🙏 


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/57b4c461-9d83-4597-ae89-407a1eaf4c46) or assign @oscarj007 to this issue.

**Comment by Power70** [comment]
@Power70 has applied to work on this issue as part of the **Stellar Wave Program**'s 5th wave.



> Hi Maintainer,
> 
> I can work on this issue
> 
> Kindly assign it to me
> 
> Thank you.


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/57b4c461-9d83-4597-ae89-407a1eaf4c46) or assign @Power70 to this issue.

**Comment by ritik4ever** [comment]
@ritik4ever has applied to work on this issue as part of the **Stellar Wave Program**'s 5th wave.



> Hi
> I’d like to work on this issue if it’s available. I’m a full-stack Web2 & Web3 developer with solid open-source experience and can start working on it right away.


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/57b4c461-9d83-4597-ae89-407a1eaf4c46) or assign @ritik4ever to this issue.
